federation/client/clientset_generated/internalclientset/clientset.go

```diff
@@ -0,0 +1,81 @@
+/*
+Copyright 2016 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package internalclientset
+
+import (
+	"github.com/golang/glog"
+	unversionedfederation "k8s.io/kubernetes/federation/client/clientset_generated/internalclientset/typed/federation/unversioned"
+	restclient "k8s.io/kubernetes/pkg/client/restclient"
+	discovery "k8s.io/kubernetes/pkg/client/typed/discovery"
+)
+
+type Interface interface {
+	Discovery() discovery.DiscoveryInterface
```

Comment:
Hi, @jianhuiz , in cluster-controller and scheduler, we will use this client interface to operate resource. Except cluster and sub-replicaset , we also need relicaset operation. So would you like to define the interface like this?

```
type Interface interface {
	Discovery() discovery.DiscoveryInterface
	Core() unversionedcore.CoreInterface
	Federation() unversionedfederation.FederationInterface
}
```